### PR TITLE
fix: use hasMounted to render calcom versionm

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes Hydration error in /event-types page

Ref: https://www.joshwcomeau.com/react/the-perils-of-rehydration/#the-solution

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by PeerRich** [comment]
i bet there is a better workaround. if not we should move this into a standalone component to prevent repeating code

**Comment by PeerRich** [comment]
i think by turning the constant into a react component, it would not have hydration issues. wanna try that?

**Comment by PeerRich** [comment]
```
export default function Credits() {
const CalComVersion = `v.${pkg.version}-${
  process.env.NEXT_PUBLIC_WEBSITE_URL === "https://cal.com" ? "h" : "sh"
}-${process.env.CALCOM_LICENSE_KEY === "" ? "ee" : "ce"}`;

 return <small className="mx-3 mt-1 mb-2 hidden text-[0.5rem] opacity-50 lg:block">
            &copy; {new Date().getFullYear()} {COMPANY_NAME} {CalComVersion}
 </small>
}
```

**Review by PeerRich — CHANGES_REQUESTED** [review]
lets turn this into a component `<Credits>`

https://github.com/calcom/cal.com/pull/6239#issuecomment-1369117119

**Review by PeerRich — APPROVED** [review]
made some changes, created `<Credits>`

File: apps/web/pages/_app.tsx [hunk 1]
```diff
@@ -44,10 +43,6 @@ function MyApp(props: AppProps) {
       <Head>
         <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
       </Head>
-      <Script
-        id="calcom-version"
-        dangerouslySetInnerHTML={{ __html: `window.CalComVersion = '${CalComVersion}'` }}
-      />
       {getLayout(
```
Comment (PeerRich) [inline apps/web/pages/_app.tsx:46]: @hariombalhara i'm removing this until we need it

**Review by PeerRich — APPROVED** [review]

File: apps/web/pages/_app.tsx (hunk 1 above)
Comment (hariombalhara) [reply apps/web/pages/_app.tsx:45]: Cool

**Pull request merged**